src/rules/lambda/LambdaLatestVersion.ts

```diff
@@ -0,0 +1,75 @@
+/*
+Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+SPDX-License-Identifier: Apache-2.0
+*/
+import { parse } from 'path';
+import { CfnFunction, Runtime } from '@aws-cdk/aws-lambda';
+import { CfnResource } from '@aws-cdk/core';
+import { NagRuleCompliance, NagRules } from '../../nag-rules';
+
+/**
+ * Lambda functions are configured to use the latest runtime version
+ * @param node the CfnResource to check
+ */
+export default Object.defineProperty(
+  (node: CfnResource): NagRuleCompliance => {
+    if (node instanceof CfnFunction) {
+      const runtime = NagRules.resolveIfPrimitive(node, node.runtime);
+      if (!runtime) {
+        // Runtime is not required for container lambdas, in this case, not applicable
+        return NagRuleCompliance.NOT_APPLICABLE;
+      }
+
+      const exp = /([a-z]+)(\d+(\.?\d+|\.x)?)?/;
+      const m = runtime.match(exp);
+
+      if (!m) {
+        // shouldn't happen, but if for some reason it does, we'll ignore the check
+        // CloudFormation will likely fail when trying to create the lambda
+        return NagRuleCompliance.NOT_APPLICABLE;
+      }
+
+      const runtimeFamily = m[1];
+      if (runtimeFamily === 'provided') {
+        return NagRuleCompliance.NOT_APPLICABLE;
+      }
+
+      // We'll pull the versions which CDK knows about to ensure we don't throw complaints
+      // about a runtime version which isn't available for use in the users CDK library.
+      const familyVersions = Runtime.ALL.filter(
+        (rt) => rt.toString().indexOf(runtimeFamily) === 0
+      )
+        .map((rt) => {
+          let match = rt.toString().match(exp);
+          return {
+            value: rt.toString(),
+            family: match![1],
+            version: parseFloat(match![2]),
+          };
+        })
+        .sort((a, b) => {
+          if (a.version < b.version) return -1;
+          else if (a.version > b.version) return 1;
+          else return 0;
+        });
+
+      if (familyVersions.length === 0) {
+        // shouldn't happen, but if for some reason it does, we'll ignore the check
+        // CloudFormation will likely fail when trying to create the lambda
+        return NagRuleCompliance.NOT_APPLICABLE;
+      }
```

Comment:
Rather than saying this is Not Applicable (since that means this rule won't appear in reports for a particular Lambda function). I'd rather throw an error here and have it bubble up to be a `CdkNagValidationFailure` so the user can debug further if they choose. Tests (and possible how we run them) will also need to be adjusted for this. [Example](https://github.com/cdklabs/cdk-nag/blob/e081e7dfd11236f415311631a108028c0a1250c5/test/Engine.test.ts#L837)

```suggestion
      if (familyVersions.length === 0) {
        throw Error(
          `Unable to find families for Lambda runtime "${runtime}", therefore the rule could not be validated.`
        )
      }
```